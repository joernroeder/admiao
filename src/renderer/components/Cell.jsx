import styled from '@emotion/styled'
import { Box } from 'reflexbox'

import gridOffset from '../styled-props/gridOffset'
import gridColumn from '../styled-props/gridColumn'

const Cell = styled(Box)`
	${gridOffset};
	${gridColumn};
`

Cell.defaultProps = {
	px: 1,
}

export default Cell
