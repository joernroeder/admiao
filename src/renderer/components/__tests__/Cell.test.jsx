import React from 'react'
import { render } from '@testing-library/react'

import Cell from '../Cell'

describe('Cell component', () => {
	it('should correctly have 1/2 of the grid gutter as horizontal padding', () => {
		const { container } = render(<Cell />)
		const cell = container.firstChild

		expect(cell).toHaveStyleRule(
			'padding-left',
			global.halfGridGutterInPixel
		)
		expect(cell).toHaveStyleRule(
			'padding-right',
			global.halfGridGutterInPixel
		)

		expect(cell).toMatchSnapshot()
	})
})
