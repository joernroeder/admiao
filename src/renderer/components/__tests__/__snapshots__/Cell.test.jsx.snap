// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Cell component should correctly have 1/2 of the grid gutter as horizontal padding 1`] = `
.emotion-0 {
  box-sizing: border-box;
  margin: 0;
  min-width: 0;
  padding-left: 4px;
  padding-right: 4px;
}

<div
  class="emotion-0"
/>
`;
